hw/hw17/html_render.py

```diff
@@ -5,7 +5,109 @@
 
 """
 
+
 # The start of it all:
 # Fill it all in here.
 class Element(object):
-    pass
+
+    BASE_INDENT = "    "
+
+    def __init__(self, tag="", content=None, oneLineFlag=False,
+                 selfCloseFlag=False, **kwargs):
+        self.tag = tag
+        self.content = []
+        self.append(content)
+        self.oneLineFlag = oneLineFlag
+        self.selfCloseFlag = selfCloseFlag
+        self.attrs = []
+        self.attrStr = ''
+        if (kwargs is not None):
+            for key, val in kwargs.items():
+                self.attrs.append("%s=\"%s\"" % (key, val))
+        if (len(self.attrs) > 0):
+            self.attrStr = " " + " ".join(self.attrs)
+
+    def append(self, content):
+        if (content):
+            self.content.append(content)
+
+    def render(self, fh, indent=""):
```

Comment:
Nice render function, you're really getting a lot done here. I did notice you have trailing newline characters at the end of each output file. How would you fix that?